### PR TITLE
mesa: Add missing dependencies

### PR DESCRIPTION
**Pull request by michaelkuhn**
@chuckatkins

### CODE REVIEW

File: var/spack/repos/builtin/packages/mesa/package.py [hunk 1]
```diff
@@ -27,12 +27,15 @@ class Mesa(MesonPackage):
 
     depends_on('meson@0.45:', type='build')
     depends_on('binutils', type='build')
+    depends_on('bison', type='build')
+    depends_on('flex', type='build')
```
Comment (chuckatkins) [inline var/spack/repos/builtin/packages/mesa/package.py]: I see the bison and flex dependency by where is the gettext dependency coming in?

Comment (michaelkuhn) [reply var/spack/repos/builtin/packages/mesa/package.py]: The build was looking for/tried to call `xgettext`. On second thought, this probably should also be a build dependency then.

**Comment by chuckatkins** [comment]
I'm good with this save the change to gettext as build-only (so long as that's correct anyways).  I'll be happy to merge it once that change is made.

File: var/spack/repos/builtin/packages/mesa/package.py (hunk 1 above)
Comment (chuckatkins) [reply var/spack/repos/builtin/packages/mesa/package.py]: Agreed

**Comment by michaelkuhn** [comment]
> I'm good with this save the change to gettext as build-only (so long as that's correct anyways). I'll be happy to merge it once that change is made.

Done :-)

**Pull request merged**